concordium-contracts-common-derive/src/derive.rs

```diff
@@ -1305,15 +1322,35 @@ pub fn schema_type_derive_worker(input: TokenStream) -> syn::Result<TokenStream>
     let ast: syn::DeriveInput = syn::parse(input)?;
 
     let data_name = &ast.ident;
+    let container_attributes = ContainerAttributes::try_from(ast.attrs.as_slice())?;
 
     let body = match ast.data {
         syn::Data::Struct(ref data) => {
-            let fields_tokens = schema_type_fields(&data.fields)?;
-            quote! {
-                concordium_std::schema::Type::Struct(#fields_tokens)
+            if container_attributes.transparent {
+                if data.fields.len() != 1 {
+                    return Err(syn::Error::new(
+                        ast.span(),
+                        "'transparent' attribute can only be used on a struct with a single field",
+                    ));
+                }
+
+                // Safe to unwrap below since we already checked then length is one.
+                let field = data.fields.iter().next().unwrap();
+                schema_type_field_type(field)?
```

Comment:
```suggestion
                match data.fields.split_first() {
                    Some((field, rest)) if rest.is_empty() => {
                         schema_type_field_type(field)
                    _ => {
                    return Err(syn::Error::new(
                        ast.span(),
                        "'transparent' attribute can only be used on a struct with a single field",
                    ));
                }
```

Reply:
I tried this just now, but `data.fields` does not have a method `split_first` and cannot become a slice easily.
https://docs.rs/syn/1.0.109/syn/enum.Fields.html

Reply:
I think I looked at the wrong type then.